govcd/vm_test.go

```diff
@@ -1611,3 +1611,50 @@ func (vcd *TestVCD) Test_AddNewEmptyVMWithVmComputePolicyAndUpdate(check *C) {
 	_, err = adminVdc.SetAssignedComputePolicies(types.VdcComputePolicyReferences{VdcComputePolicyReference: beforeTestPolicyReferences})
 	check.Assert(err, IsNil)
 }
+
+func (vcd *TestVCD) Test_VMUpdateStorageProfile(check *C) {
+	config := vcd.config
+	if config.VCD.StorageProfile.SP1 == "" || config.VCD.StorageProfile.SP2 == "" {
+		check.Skip("Skipping test because no both storage profile has to be configured")
+	}
+
+	vapp, err := createVappForTest(vcd, "Test_VMUpdateStorageProfile")
+	check.Assert(err, IsNil)
+	check.Assert(vapp, NotNil)
+
+	cat, err := vcd.org.GetCatalogByName(vcd.config.VCD.Catalog.Name, true)
+	check.Assert(err, IsNil)
+	check.Assert(cat, NotNil)
+
+	var storageProfile types.Reference
+
+	storageProfile, _ = vcd.vdc.FindStorageProfileReference(vcd.config.VCD.StorageProfile.SP1)
+
+	createdVm, err := makeEmptyVm(vapp, "Test_VMUpdateStorageProfile")
+	check.Assert(err, IsNil)
+	check.Assert(createdVm, NotNil)
+	AddToCleanupList("Test_VMUpdateStorageProfile", "vm", "", "Test_VMUpdateStorageProfile")
```

Comment:
This is redundant. In fact, it leads to an error during cleanup. The parent vApp is already scheduled for deletion within `createVappFortest`. When it comes to this VM, the deletion fails because the parent vApp has already disappeared.

Reply:
removed